src/modules/rppt_tensor_filter_augmentations.cpp

```diff
@@ -25,12 +25,90 @@ SOFTWARE.
 #include "rppdefs.h"
 #include "rppi_validate.hpp"
 #include "rppt_tensor_filter_augmentations.h"
+#include "cpu/host_tensor_filter_augmentations.hpp"
 
 #ifdef HIP_COMPILE
 #include <hip/hip_fp16.h>
 #include "hip/hip_tensor_filter_augmentations.hpp"
 #endif // HIP_COMPILE
 
+/******************** sobel_filter ********************/
+
+RppStatus rppt_sobel_filter_host(RppPtr_t srcPtr,
+                                 RpptDescPtr srcDescPtr,
+                                 RppPtr_t dstPtr,
+                                 RpptDescPtr dstDescPtr,
+                                 Rpp32u sobelType,
+                                 Rpp32u kernelSize,
+                                 RpptROIPtr roiTensorPtrSrc,
+                                 RpptRoiType roiType,
+                                 rppHandle_t rppHandle)
+{
+    if ((kernelSize != 3) && (kernelSize != 5) && (kernelSize != 7))
+        return RPP_ERROR_INVALID_ARGUMENTS;
+    if ((sobelType != 0) && (sobelType != 1) && (sobelType != 2))
+        return RPP_ERROR_INVALID_ARGUMENTS;
+
+    // convert image to grey scale if input is RGB image
+    RppPtr_t tempPtr = srcPtr;
+    if (srcDescPtr->c == 3)
+    {
+        RpptSubpixelLayout srcSubpixelLayout = RpptSubpixelLayout::RGBtype;
+        tempPtr = rpp::deref(rppHandle).GetInitHandle()->mem.mcpu.scratchBufferHost;
+        rppt_color_to_greyscale_host(srcPtr, srcDescPtr, tempPtr, dstDescPtr, srcSubpixelLayout, rppHandle);
+    }
+
+    if ((srcDescPtr->dataType == RpptDataType::U8) && (dstDescPtr->dataType == RpptDataType::U8))
+    {
+        sobel_filter_host_tensor(static_cast<Rpp8u*>(tempPtr) + srcDescPtr->offsetInBytes,
+                                 dstDescPtr,
+                                 static_cast<Rpp8u*>(dstPtr) + dstDescPtr->offsetInBytes,
+                                 dstDescPtr,
```

Comment:
If user gives a dstDescPtr->c as 3 do we have a check?

Reply:
Added a check now
https://github.com/sampath1117/rpp/blob/sr/sobel_filter_host/src/modules/rppt_tensor_filter_augmentations.cpp#L51